Singularity "Warp" Shader

jamessimo opened the pull request:
### Purpose
This adds a "warp" effect around the singularity using a grabpass shader on the prefab. This is to make the singularity look like its really beanding space time and has the nice effect of objects getting sucked in! 
![image](https://user-images.githubusercontent.com/4888377/144867018-1da7c1df-e2b7-4ca1-9170-915117516b4a.png)


### Notes:
The Singularity has 2 shaders on it. One applies to the "lit" layers and the other only applies on the water or stars layer.
Added a check to mouse utils to make sure the texture can be read. 
Added some other WIP shaders under shaders/experimental. These are not used anywhere but are being worked on. 
Effects may need to be tweaked as we get some user feedback.

### Changelog:
CL: [Improvement] Singularity now looks like its bending space time using shaders!

jamessimo committed:
First pass at warp shader

uses grabpass method
added minor mouse check to not read shader textrure
Singu prefab added warp textures

jamessimo committed:
Minor improvements to warp shader + organized unused shaders into experimental

also removed debug logs

jamessimo committed:
Update Singularity.cs

quick fix bad var not showing effect

jamessimo committed:
Better effect scaling

Now takes into account the singu's scale when applying effects and removed ugly multi function calls.

NoooneyDude reviewed (commented):
Your `UpdateWarpFX` method will only be called from the server (aside from initial spawn via Awake / OnEnable). So the scale won't be synced. You can probably get away with just moving the method call to the `SyncDynamicScale` method.

Did you test with a client / server? Testing in editor means the client _is_ the server, so it'll appear to work.

jamessimo committed:
Made effect sync with server + fix light

also made tweaks to effect values and adding clamps to stop some values getting too big due to lob-sided scaling

ThatDan123 commented on UnityProject/Assets/Scripts/Objects/Singularity.cs:
```diff
@@ -168,6 +168,7 @@ private void SyncDynamicScale(Vector3 oldScale, Vector3 newScale)
 			}
 
 			gameObject.transform.LeanScale(newScale, updateFrequency);
+			UpdateWarpFX(CurrentStage);
```
current stage isnt sync'd

ThatDan123 replied:
You could link the `scaledRadius` and `scaledEffect` by the `Vector3 newScale` was nooneys idea. 

Or could add sync var to stage and then add new hook for it.

jamessimo committed:
Added better hooks + better scaling logiv

+ added new lighting
+  enhanced scaling logic

ThatDan123 commented on UnityProject/Assets/Scripts/Objects/Singularity.cs:
```diff
@@ -209,6 +230,8 @@ private void SingularityUpdate()
 			// will sync to clients
 			dynamicScale = GetDynamicScale();
 
+			UpdateWarpFX(CurrentStage);
```
```suggestion
```

ThatDan123 commented on UnityProject/Assets/Scripts/Objects/Singularity.cs:
```diff
@@ -153,6 +168,12 @@ private void SyncDynamicScale(Vector3 oldScale, Vector3 newScale)
+
+		private void SyncCurrentStage(SingularityStages oldStage, SingularityStages newStage)
+		{
+			currentStage = newStage;
```
```suggestion
			currentStage = newStage;
			
			UpdateWarpFX(CurrentStage);
```

jamessimo committed:
Update Singularity.cs

Daniels suggestion
